CaloMC/src/CaloDigiFromShower_module.cc

```diff
@@ -413,82 +413,83 @@ namespace mu2e {
+      std::cout<<std::endl;
   }
   void CaloDigiFromShower::diag1(int iRO, double time, size_t peakP, const std::vector<int>& wf, int ped)
   {
-     std::cout<<"Created caloDigi with roID = "<<iRO<<"  t0="<<time<<" peak="<<peakP<<" and content ";
-     for (const auto  &v : wf) {if (v-ped >=minPeakADC_ ) std::cout<< "**"; std::cout<<v-ped<<" ";}
-     std::cout<<std::endl;
+      std::cout<<"Created caloDigi with roID = "<<iRO<<"  t0="<<time<<" peak="<<peakP<<" and content ";
+      for (const auto  &v : wf) {if (v-ped >=minPeakADC_ ) std::cout<< "**"; std::cout<<v-ped<<" ";}
+      std::cout<<std::endl;
   }
   void CaloDigiFromShower::diag2(const CaloDigiCollection& caloDigiColl)
   {      
-     std::map<int,double> enerMap;
-     for (const auto& digi : caloDigiColl) enerMap[digi.roId()] += digi.waveform().at(digi.peakpos())/MeVToADC_;
-     std::cout<<"[CaloDigiFromShower] energy equivalent per RoID"<<std::endl;
-     for (auto& kv : enerMap) std::cout<<" roID: "<<kv.first<<"   Ener: "<<kv.second<<std::endl;
+      std::map<int,double> enerMap;
+      for (const auto& digi : caloDigiColl) enerMap[digi.roId()] += digi.waveform().at(digi.peakpos())/MeVToADC_;
+      std::cout<<"[CaloDigiFromShower] energy equivalent per RoID"<<std::endl;
+      for (auto& kv : enerMap) std::cout<<" roID: "<<kv.first<<"   Ener: "<<kv.second<<std::endl;
   }
 
 
 
   //-------------------------------------------------------------------------------------------------------------------
   void CaloDigiFromShower::plotNoise(const std::vector<std::vector<double>>& waveforms, const std::vector<int>& pedestals)
   {      
-      const unsigned nGr = waveforms.size();
-      TGraph gr[nGr];
-      
-      double maxe(0);
-      for (unsigned i=0;i<waveforms.size();++i)
-      {
-         const auto& wf = waveforms[i];
-         double xv[9999],yv[9999];
-         for (unsigned j=0;j<wf.size();++j) {xv[j]=j*digiSampling_+0.5*digiSampling_;yv[j]=wf[j]-pedestals[i]; maxe=std::max(maxe,std::abs(yv[j]));}
-         gr[i] = TGraph(wf.size(),xv,yv);
-         gr[i].SetLineColor(i%20+1);
-      }
-      
-      static int nplots(0);
-      std::stringstream ss;ss<<"noise_"; ss<<nplots;ss<<".pdf";
-      ++nplots;
-
-      gStyle->SetOptStat(0);
-      TCanvas c1("c1","c1");
-      TH1F empty("e","waveform",10,0,waveforms[0].size()*digiSampling_);
-      empty.GetYaxis()->SetRangeUser(-maxe,maxe);
-      empty.GetXaxis()->SetTitle("time [ns]");
-      empty.GetYaxis()->SetTitle("ADC");
-      empty.Draw();
-      for (unsigned i=0;i<std::min(100u,nGr);++i) gr[i].Draw("L");
-      c1.SaveAs(ss.str().c_str());   
+       const unsigned nMaxPlot(100u);
+       const unsigned nGr = waveforms.size();
+       TGraph gr[nGr];
+
+       double maxe(0);
+       for (unsigned i=0;i<waveforms.size();++i)
+       {
+          const auto& wf = waveforms[i];
+          double xv[9999],yv[9999];
```

Comment:
What determines these sizes?  is there a risk of overflow?  Note, you can extract the array content of a vector with .data(), to use in TGraph.

Reply:
Ah, I learned something (very useful in fact). Modified the code to use .data() and remove the need for the arrays.